src/map/lua/luautils.cpp

```diff
@@ -5724,4 +5732,47 @@ namespace luautils
 
         return CLuaBaseEntity(PEntity);
     }
+
+    // Fishing Contest utilities
+    auto GetFishingContest() -> sol::table
+    {
+        sol::table table = lua.create_table();
+
+        table["status"]     = fishingcontest::GetContestStatus();
+        table["criteria"]   = fishingcontest::GetContestCriteria();
+        table["measure"]    = fishingcontest::GetContestMeasure();
+        table["fishid"]     = fishingcontest::GetContestFish();
+        table["starttime"]  = fishingcontest::GetContestStartTime();
+        table["changetime"] = fishingcontest::GetContestChangeTime();
+
+        return table;
+    }
+
+    void InitNewFishingContest()
+    {
+        // This function will destroy the current contest and start a new one
+        fishingcontest::InitNewContest();
+    }
+
+    void SetContestParameters(uint16 fishId, uint8 measure, uint8 criteria)
+    {
+        // Note: Validation of fish item id should be done in lua, since the fish table is also in lua
+        fishingcontest::SetContestFish(fishId);
+        fishingcontest::SetContestMeasure((FISHING_CONTEST_MEASURE)std::min((uint8)measure, (uint8)CONTEST_MEASURE_SMALLEST));
+        fishingcontest::SetContestCriteria((FISHING_CONTEST_CRITERIA)std::min((uint8)criteria, (uint8)CONTEST_CRITERIA_BOTH));
+    }
+
+    void ProgressFishingContest()
+    {
+        // This function will get called every tick when automatic progression is enabled
+        fishingcontest::ProgressContest();
+    }
+
+    void InitializeFishingContestSystem()
+    {
+        // IMPORTANT: This should only be called on the Zone Init in Selbina
```

Comment:
Could you add safety checks for this (and logging) to make sure this is enforced?